core/src/main/java/com/graphhopper/reader/dem/RoadElevationInterpolator.java

```diff
@@ -0,0 +1,48 @@
+package com.graphhopper.reader.dem;
+
+import com.graphhopper.util.Helper;
+import com.graphhopper.util.PointList;
+
+/**
+ * This class smooths the elevation data of a PointList by calculating the average elevation over
+ * multiple points of that PointList.
+ * <p>
+ * The ElevationData is read from rectangular tiles. Especially when going along a cliff,
+ * valley, or pass, it can happen that a small part of the road contains incorrect elevation data.
+ * This is because the elevation data is coarse and sometimes contains errors.
+ * <p>
+ * This can lead to incorrect ascend, descend, and distance calculation of a route.
+ *
+ * @author Robin Boldt
+ */
+public class RoadElevationInterpolator {
+
+    private final static int MIN_GEOMETRY_SIZE = 3;
+    // The max amount of points to go left and right
+    private final static int MAX_SEARCH_RADIUS = 8;
+    // If the point is farther then this, we stop averaging
+    private final static int MAX_SEARCH_DISTANCE = 180;
+
+    public static PointList smoothElevation(PointList geometry) {
+        if (geometry.size() >= MIN_GEOMETRY_SIZE) {
```

Comment:
Yes true, I had it there because I was experimenting with different min geometry sizes to see if there is a difference. Also when I played with the LOESS regression, this required at least 7? points to work.